index.js

```diff
@@ -357,13 +357,21 @@ class ClientServiceBase {
    */
   async validateLedger(assetId, startAge = ClientServiceBase.minAge,
       endAge = ClientServiceBase.maxAge) {
-    const properties = new ClientProperties(this.properties, [], []);
+    if (!(endAge >= startAge && startAge >= ClientServiceBase.minAge &&
+          endAge <= ClientServiceBase.maxAge)) {
+      throw new ClientError(
+          StatusCode.CLIENT_RUNTIME_ERROR,
+          'invalid ages are specified',
+      );
+    }
```

Comment:
This code snippet is moved from a lower layer in function `_createLedgerValidationRequest` because we can validate the arguments for both the non-linearizable and the linearizable ledger validation with the same logic.